frontend/app/routes/protected/letters/index.tsx

```diff
@@ -45,17 +45,18 @@ export async function loader({ context: { appContainer, session }, params, reque
   const userInfoToken: UserinfoToken = session.get('userInfoToken');
   invariant(userInfoToken.sin, 'Expected userInfoToken.sin to be defined');
 
-  const clientNumber =
-    session.find('clientNumber').unwrapUnchecked() ??
-    (await appContainer.get(TYPES.domain.services.ApplicantService).findClientNumberBySin({
+  const clientNumberOption = session.find('clientNumber').or(
+    await appContainer.get(TYPES.domain.services.ApplicantService).findClientNumberBySin({
       sin: userInfoToken.sin,
       userId: userInfoToken.sub,
-    }));
+    }),
+  );
```

Comment:
Using `or` will execute `findClientNumberBySin` to set his `optb` param. `findClientNumberBySin` must only be called if `clientNumber` is not in the session.

```suggestion
  const clientNumberOption: Option<string> = session.has('clientNumber')
    ? session.find('clientNumber')
    : await appContainer.get(TYPES.domain.services.ApplicantService).findClientNumberBySin({
        sin: userInfoToken.sin,
        userId: userInfoToken.sub,
      });
```